c7n/resources/iam.py

```diff
@@ -104,6 +104,42 @@ def get_resources(self, resource_ids, cache=True):
         return resources
 
 
+@Role.action_registry.register('tag')
+class RoleTag(Tag):
+    """Tag an iam role."""
+
+    permissions = ('iam:TagRole',)
+
+    def process_resource_set(self, client, roles, tags):
+        for role in roles:
+            try:
+                client.tag_role(RoleName=role['RoleName'], Tags=tags)
+            except client.exceptions.NoSuchEntityException:
+                continue
+
+
+@Role.action_registry.register('remove-tag')
+class RoleRemoveTag(RemoveTag):
+    """Remove tags from an iam role."""
+
+    permissions = ('iam:UntagRole',)
+
+    def process_resource_set(self, client, roles, tags):
+        for role in roles:
+            try:
+                client.untag_role(RoleName=role['RoleName'], TagKeys=tags)
+            except client.exceptions.NoSuchEntityException:
+                continue
+
+
+@Role.action_registry.register('mark-for-op')
+class RoleTagDelayedAction(TagDelayedAction):
```

Comment:
instead of defining an empty class could be registered as just a function per marked-for-op below

Reply:
the tag machinery was refactored since the iam user tagging support, to make subclasses of mark-for-op extraneous.